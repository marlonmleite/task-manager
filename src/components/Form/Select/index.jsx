import PropTypes from 'prop-types'
import React from 'react'
import withFormGroup from '../withFormGroup'
import SelectStyled from './styled'

export const SelectComponent = props => (
  <SelectStyled {...props} className={!props.value ? 'empty' : ''} />
)

SelectComponent.defaultProps = {
  value: '',
  multiple: false,
}

SelectComponent.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
  ]),
  multiple: PropTypes.bool,
}

export const Select = withFormGroup(SelectComponent)
