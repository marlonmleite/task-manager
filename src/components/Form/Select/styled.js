import styled from 'styled-components'
import { InputComponent } from '../Input'

const Select = InputComponent.withComponent('select')

const SelectStyled = styled(Select)`
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
`

export default SelectStyled
