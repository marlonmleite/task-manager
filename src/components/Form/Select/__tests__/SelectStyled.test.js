import React from 'react'
import { shallow } from 'enzyme'
import Select from '../styled'

describe('SelectStyled', () => {
  it('should render', () => {
    const select = shallow(<Select />)

    expect(select).toMatchSnapshot()
  })
})
