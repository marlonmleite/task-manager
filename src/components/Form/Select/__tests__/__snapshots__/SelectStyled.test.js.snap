// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectStyled should render 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<styled.select
  className="c0"
  type="text"
/>
`;
